proto/opamp.proto

```diff
@@ -348,6 +356,37 @@ message OtherConnectionSettings {
     // Other connection settings. These are Agent-specific and are up to the Agent
     // interpret.
     map<string, string> other_settings = 4;
+
+    // Optional connection specific TLS settings.
+    // Status: [Development]
+    TLSConnectionSettings tls = 5;
+}
+
+// Status: [Development]
+message TLSConnectionSettings {
+  // Insecure is false by default, if true TLS will be disabled for the connection.
+  bool insecure = 1;
+
+  // Optional path to the CA file on disk
+  string ca_file = 2;
+
+  // Alternative to ca_file, Provides CA cert contents as a string
+  string ca_pem = 3;
```

Comment:
How do we expect this to work? Is OpAMP servers expected to know where on agent's disk these files are located? How would the server know?

This also potentially is an attack vector where malicious server can direct the agent to read an arbitrary file. On its own it may seem benign but if we end up having another weakness where the file that was read for some reasons gets reported to the server this will allow the server to remotely read arbitrary files from agent's machine. This goes against our zero trust [recommendations](https://github.com/open-telemetry/opamp-spec/blob/main/specification.md#general-recommendations) in the Security section.

Should we instead send the _content_ of CA certificate to the agent to use? The agent can save the content locally and use it in subsequent connections.

This can work particularly well for `TelemetryConnectionSettings` and `OtherConnectionSettings`, the idea being that OpAMP server has the knowledge about the CA used by telemetry and other connection destinations.

And even for `OpAMPConnectionSettings` this may make sense as a flow, where the agent first connects without the TLS certificate, in sort of a bootstraping mode, then accepts server offered CA certificate for future verifications. This is similar to Trust On First Use flow that we already have but in the opposite direction (trust of server, instead of trust of client).

If we think this is a valid usage of TLSConnectionSettings I think we should add a corresponding section in the spec and include a sequence diagram that explains it.

Reply:
When specifying the file path we assume it is present at that location across all agents. It can be provisioned by some other tool (baked into the image, or by a configuration management tool).

But I can see how having a filepath specification goes against our current recommendations. It's specified because it is a common approach to TLS configuration, however I can remove the it if we want to keep our security exposure low.

The content can instead be specified with the `ca_pem` attribute (I'll rename to `ca_pem_content` to make that clearer).

I like the idea of a TOFU like mechanism for accepting CAs from servers, I'll try to add the description for this in the spec.
At elastic we do something a little similar to that, we allow the specification of a CA by a fingerprint hash and add the matching CA to the cert pool when it is found in a connection ([implementation here](https://github.com/elastic/elastic-agent-libs/blob/main/transport/tlscommon/tls_config.go#L177)). This assumes that what ever is installing an agent has some additional information about the server and it prevents an agent from blindly trusting any CA. This may be worth breaking out into another issue/pr, what do you think?

Reply:
>however I can remove the it if we want to keep our security exposure low.

Yes, I think we should remove it for now.